### PR TITLE
Strengthen PR review comment processing: pass file/line context to Claude

### LINKED ISSUE #374: Strengthen PR review comment processing: pass file/line context to Claude
## Problem

When Fabrik re-invokes the Review stage comment handler to address PR review thread (inline) comments, it passes only the comment body, author, and timestamp in the prompt. Missing from the prompt:

- The file path the comment targets
- The line number(s) the comment is attached to
- The code diff hunk the comment was made on
- Any prior comments in the same thread (for threaded conversations)
- The thread's resolved/unresolved state

This makes it hard for Claude to address line-scoped feedback from reviewers like GitHub Copilot (whose comments are always inline, e.g. "fix error handling here" attached to a specific line). Claude has to infer where the change belongs from the comment body alone, which is error-prone.

## Summary

Extend the GraphQL fetch, `commentNodeData` and `Comment` structs, `buildCommentReviewPrompt`, and the `fabrik-review-comment` skill to include file path, line number, and diff hunk context when processing PR review thread comments. This gives Claude the exact location and surrounding code to apply reviewer feedback correctly.

### Extend the GraphQL fetch

Update the `reviewThreads` block in `FetchItemDetails` (`github/project.go`) to pull thread-level and comment-level location fields:

```graphql
reviewThreads(first: 50) {
  nodes {
    id
    isResolved
    path              # NEW — file path at thread level
    line              # NEW — current-diff line at thread level
    originalLine      # NEW — base-diff line at thread level
    diffSide          # NEW — LEFT or RIGHT side of the diff
    comments(first: 20) {
      nodes {
        id
        databaseId
        author { login }
        body
        createdAt
        diffHunk        # NEW — ~3-line surrounding code context
        path            # NEW — file path at comment level
        line            # NEW — current-diff line at comment level
        originalLine    # NEW — base-diff line at comment level
        reactionGroups { content reactors { totalCount } }
      }
    }
  }
}
```

### Extend `commentNodeData`

Add location fields to the `commentNodeData` struct (`github/project.go`) so the new GraphQL fields are decoded:
- `DiffHunk string` (`json:"diffHunk"`)
- `Path string` (`json:"path"`)
- `Line int` (`json:"line"`)
- `OriginalLine int` (`json:"originalLine"`)

### Update the `Comment` struct

Add location fields to `github.Comment` (`github/types.go`):
- `Path string` — file path (populated for review thread comments)
- `Line int` — line number in the current diff (0 when not applicable)
- `OriginalLine int` — line number in the original base (fallback when `Line` is 0)
- `DiffHunk string` — the diff context hunk around the comment

These fields are empty for regular issue and PR body comments; they are populated only when converting review thread comment nodes via `toComment`.

### Update the GraphQL response struct

Extend the anonymous struct for review thread nodes inside `FetchItemDetails` to decode `path`, `line`, `originalLine`, and `diffSide` at the thread level. The `diffSide` is thread-level metadata; it is not stored in `Comment` but may be forwarded to the prompt for context.

### Enrich `buildCommentReviewPrompt`

When the comments being processed include review thread comments (i.e., `Comment.Path != ""`), format each comment block to include:
- File path
- Line number (prefer `Line`; fall back to `OriginalLine` when `Line` is 0)
- Diff hunk (fenced code block with the hunk content)
- Thread ID (so Claude can reason about which feedback belongs together)

Example rendered format for one review thread comment:

```
**@copilot** (2026-01-15 10:30) [Thread: RT_abc123]
**File:** `engine/claude.go` **Line:** 243
**Diff context:**
```diff
@@ -241,7 +241,7 @@
-	prompt := buildCommentReviewPrompt(stage, issue, comments)
+	prompt := buildCommentReviewPrompt(stage, issue, comments, opts)
```
Please fix the error handling here — the returned error is silently dropped.
```

Comments without location fields (regular issue/PR comments) continue to render as before.

### Update the `fabrik-review-comment` skill

Add a section explaining that PR review thread comments include location context, and instruct Claude to:
- Use the file path and line number to navigate directly to the relevant location
- Use the diff hunk to understand the exact code being reviewed before making changes
- Address each thread's feedback at the correct file location
- Use `gh api /repos/{owner}/{repo}/pulls/{pr}/comments` as a fallback if more context is needed

## Scope

**In scope:**
- GraphQL query extension for location fields on review thread comments
- `commentNodeData` and `Comment` struct additions
- `toComment` conversion populating the new fields from comment-level data
- `buildCommentReviewPrompt` enrichment for review thread comments with location context
- `fabrik-review-comment` skill update
- Unit tests for prompt enrichment and struct population

**Out of scope:**
- Posting Fabrik replies into the review thread (separate issue)
- Tracking resolution of individual threads during Claude's work (Claude should figure this out)
- Pagination for review threads beyond the current `first: 50` / `first: 20` limits
- Storing `diffSide` in the `Comment` struct (thread-level metadata, use in prompt only if useful)

## Tests

- `buildCommentReviewPrompt` with review thread comments (non-empty `Path`, `Line`, `DiffHunk`) includes file path, line number, and diff hunk in the output
- `buildCommentReviewPrompt` with regular comments (empty `Path`) renders without location fields (backwards compatibility)
- `ProjectItem.LinkedPRReviewThreadComments` carries `Path`, `Line`, `OriginalLine`, and `DiffHunk` after `FetchItemDetails` — verified via a unit test that populates the new fields from a mock GraphQL response

## Prior Art / Context

GitHub's GraphQL API has exposed `diffHunk`, `path`, `line`, and `originalLine` on `PullRequestReviewComment` nodes for some time. GitHub Copilot code review exclusively uses inline comments attached to specific lines, never top-level PR body comments. Without line context, Claude must guess where to apply Copilot's feedback from the comment body alone.

The existing `toComment` conversion in `github/project.go` already maps `commentNodeData` to `Comment` and sets `ReviewThreadID` — the new location fields extend this same mapping.

## Risks / Dependencies

- `commentNodeData` is reused across issue comments, PR comments, and review thread comments. Adding location fields is additive (zero values for non-review comments) and does not affect existing callers.
- If a review thread comment has `line: 0` (comment on a deleted line), `originalLine` should be used as fallback; both being 0 means the line reference is unavailable and the prompt should omit the line number.

### PR DESCRIPTION
**Pull request by verveguy**
Closes #374

### CODE REVIEW

**Comment by verveguy** [comment]
🏭 **Fabrik — stage: Review**
*branch: fabrik/issue-374 | commit: 81df7e53 | main: b63a062de08b8bcd888d0d545ae2641e778a1410 | 2026-04-15 00:00 UTC*

The push succeeded. Here's the complete review:

## Review Findings

### Verified

**All 11 plan tasks completed:**
- `Comment` struct extended with `Path`, `Line`, `OriginalLine`, `DiffHunk` (zero-valued for non-review-thread comments)
- `commentNodeData` extended with matching JSON-tagged location fields
- GraphQL `reviewThreads` block updated with thread-level (`path`, `line`, `originalLine`, `diffSide`) and comment-level (`diffHunk`, `path`, `line`, `originalLine`) fields
- Anonymous thread-node struct extended with `Path`, `Line`, `OriginalLine`, `DiffSide` decoder fields
- `toComment` copies all four location fields from `commentNodeData` to `Comment`
- `buildCommentReviewPrompt` correctly branches on `c.Path != ""`, applies the `Line → OriginalLine → omit` fallback, renders the fenced diff block, and includes `[Thread: ...]` in the header
- 4 tests added: `TestFetchItemDetails_ReviewThreadLocationFields`, `TestBuildCommentReviewPrompt_ReviewThreadComment`, `TestBuildCommentReviewPrompt_RegularComment_NoLocation`, `TestBuildCommentReviewPrompt_ReviewThreadComment_ZeroLine`
- `fabrik-review-comment` skill updated with a clear example and 5-point guidance section

**Test results:** All packages pass with `-race` (0 failures, 0 races)

**Build:** `go build ./...` — clean

**Vet:** `go vet ./...` — clean

**Rebase:** Branch was already current with main; force-pushed after local rebase applied.

### No issues found

The implementation is additive with no behavioral changes for non-review-thread comments. Edge cases (zero `Line`, zero `OriginalLine`, empty `DiffHunk`, empty `ReviewThreadID`) are all handled correctly and covered by tests.

Reviewed the implementation adding file/line/diff-hunk context to PR review thread comments. All 11 plan tasks are complete and correct. Build is clean, all tests pass with race detection, vet is clean. No issues found — PR is ready for human review.

---
Used 19/50 turns, 7k input / 5k output tokens.

**Comment by verveguy** [comment]
🏭 **Fabrik — stage: Validate**
*branch: fabrik/issue-374 | commit: 3850e83f | main: 42bb85c9ed3a25fc29fb241f024e3719484a4ebb | 2026-04-15 00:07 UTC*

## Validation Report

### Requirements: 5/5 passed

- [x] **GraphQL fetch extended** — `reviewThreads` block now includes `path`, `line`, `originalLine`, `diffSide` at thread level and `diffHunk`, `path`, `line`, `originalLine` at comment level (`github/project.go:347-361`)
- [x] **`commentNodeData` extended** — Added `DiffHunk string`, `Path string`, `Line *int`, `OriginalLine *int` with JSON tags (`github/project.go:26-29`); nullable int fields use `*int` per the Copilot review fix
- [x] **`Comment` struct extended** — Added `Path`, `Line`, `OriginalLine`, `DiffHunk` with explanatory comments (`github/types.go:81-92`)
- [x] **Anonymous thread-node struct extended** — Added `Path`, `Line *int`, `OriginalLine *int`, `DiffSide *string` for thread-level decoding (`github/project.go:473-476`); `DiffSide` decoded but not propagated as specified
- [x] **`toComment` updated** — Nil-safe dereference of `*int` fields, copies all four location fields to returned `Comment` (`github/project.go:601-616`)
- [x] **`buildCommentReviewPrompt` enriched** — Branches on `c.Path != ""`, renders file/line header, `Line → OriginalLine → omit` fallback, fenced diff hunk, thread ID in header (`engine/claude.go:609-634`); regular comments render unchanged
- [x] **`fabrik-review-comment` skill updated** — Added section with 5-point guidance including file/line navigation, diff hunk reading, and `gh api` fallback (`SKILL.md:37-43`)

### Test Suite: PASSED

- All packages pass with `-race` (0 failures, 0 races)
- Build: clean (`go build ./...`)
- Vet: clean (`go vet ./...`)
- New tests: `TestFetchItemDetails_ReviewThreadLocationFields`, `TestBuildCommentReviewPrompt_ReviewThreadComment`, `TestBuildCommentReviewPrompt_RegularComment_NoLocation`, `TestBuildCommentReviewPrompt_ReviewThreadComment_ZeroLine` — all pass

### Regressions: None detected

Additive change only — zero-valued `Path` means regular comments render identically to before. Backwards compatibility test (`TestBuildCommentReviewPrompt_RegularComment_NoLocation`) explicitly verifies this.

### Branch state

- 7 commits ahead of main; fully pushed to `origin/fabrik/issue-374`
- Branch is up to date with main (rebase confirmed no-op)
- Working tree clean

### Verdict: READY TO MERGE

All 11 plan tasks verified complete. Build, vet, and full test suite pass with `-race`. Requirements confirmed implemented: GraphQL query extended with location fields, `commentNodeData`/`Comment` structs extended, `toComment` populates location fields with nil-safe `*int` dereference, `buildCommentReviewPrompt` renders enriched blocks for review thread comments with correct `Line → OriginalLine → omit` fallback, and `fabrik-review-comment` skill updated with navigation guidance. No regressions detected.

---
Used 19/50 turns, 9k input / 3k output tokens.

**Comment by verveguy** [comment]
🏭 **Fabrik — stage: Validate**
*branch: fabrik/issue-374 | commit: 3850e83f | main: 42bb85c9ed3a25fc29fb241f024e3719484a4ebb | 2026-04-15 00:07 UTC*

## Validation Report

### Requirements: 5/5 passed
- [x] **GraphQL fetch extended** — `reviewThreads` block in `FetchItemDetails` now includes `path`, `line`, `originalLine`, `diffSide` at thread level and `diffHunk`, `path`, `line`, `originalLine` at comment level
- [x] **`commentNodeData` extended** — `DiffHunk string`, `Path string`, `Line *int`, `OriginalLine *int` added; nullable ints use `*int` (enhanced from spec to handle GitHub `null` explicitly)
- [x] **`Comment` struct updated** — `Path`, `Line`, `OriginalLine`, `DiffHunk` added with correct documentation; `toComment` maps all fields with nil-safe dereference
- [x] **`buildCommentReviewPrompt` enriched** — branches on `c.Path != ""`, renders file/line/thread header, applies `Line → OriginalLine → omit` fallback, renders fenced diff hunk, falls back to existing format for regular comments
- [x] **`fabrik-review-comment` skill updated** — new "PR Review Thread Comments" section with all 4 required guidance points (navigate by path/line, read diff hunk, group by thread ID, `gh api` fallback)

### Test Suite: PASSED
- 4 new tests all pass: `TestFetchItemDetails_ReviewThreadLocationFields`, `TestBuildCommentReviewPrompt_ReviewThreadComment`, `TestBuildCommentReviewPrompt_RegularComment_NoLocation`, `TestBuildCommentReviewPrompt_ReviewThreadComment_ZeroLine`
- Full suite: 10 packages, all pass with race detector
- Build: clean
- Vet: clean

### Regressions: None detected
Pre-existing tests unchanged. Changes are purely additive — zero values for non-review-thread comments; existing callers unaffected.

### Branch State: Clean
- Rebased onto latest main (already current)
- Working tree clean
- All changes pushed to `origin/fabrik/issue-374`

### Verdict: READY TO MERGE

Validation passed. All 5 requirements from the spec are fully implemented: GraphQL query extended with location fields, `commentNodeData`/`Comment` structs updated (using `*int` for nullable fields), `toComment` maps all fields with nil-safe dereference, `buildCommentReviewPrompt` enriched with file/line/hunk rendering for review thread comments with correct line fallback logic, and the `fabrik-review-comment` skill updated with a PR review thread context section. All 4 new tests pass, full suite clean with race detector, build and vet clean.

---
Used 18/50 turns, 9k input / 3k output tokens.
